lib/connectors/memory.js

```diff
@@ -54,10 +55,38 @@ Memory.prototype.getTypes = function() {
 };
 
 Memory.prototype.connect = function(callback) {
+  var self = this;
+  if (this.connected) {
+    // It's already connected or being connected
+    // Do NOT run loadFromFile again as it will override the cache
+    if (callback) process.nextTick(callback);
+    return;
+  }
+  if (callback) {
+    // Queue the callback
+    this.pendingConnectCallbacks = this.pendingConnectCallbacks || [];
```

Comment:
I started with the memory connector and then realized it should be handled at datasource level so that all connectors don't have to worry about it. I'm removing the logic from memory connector.